nan.h

```diff
@@ -1816,6 +1816,45 @@ namespace Nan {
   enum Encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER};
 }
 
+NAN_INLINE v8::Local<v8::Value> NanEncode(const void *buf, size_t len, enum Nan::Encoding encoding = Nan::BINARY) {
+#if (NODE_MODULE_VERSION > 0x000B)
+return node::Encode(v8::Isolate::GetCurrent(), buf, len, static_cast<node::encoding>(encoding));
+#else
+# if  (NODE_MODULE_VERSION < 0x000B)
+  if (encoding == Nan::BUFFER) {
+    return node::Encode(buf, len, node::BINARY);
+  }
+# endif
+  return node::Encode(buf, len, static_cast<node::encoding>(encoding));
+#endif
+}
+
+NAN_INLINE ssize_t NanDecodeBytes(v8::Handle<v8::Value> val, enum Nan::Encoding encoding = Nan::BINARY) {
+#if (NODE_MODULE_VERSION > 0x000B)
+  return node::DecodeBytes(v8::Isolate::GetCurrent(), val, static_cast<node::encoding>(encoding));
+#else
+# if (NODE_MODULE_VERSION < 0x000B)
+  if (encoding == Nan::BUFFER) {
+    return node::DecodeBytes(val, node::BINARY);
```

Comment:
`DecodeBytes(buf, BUFFER)` calls ToString() on the buffer, which interprets (for better or worse) the contents of the buffer as UTF-8.  It's arguably a bug but it's better to be compatible.